Cython/Compiler/ExprNodes.py

```diff
@@ -1746,36 +1746,25 @@ def coerce_to_boolean(self, env):
         bool_value = bool(self.value)
         return BoolNode(self.pos, value=bool_value, constant_result=bool_value)
 
+    def estimate_max_charval(self):
+        # most strings will be ASCII
+        if self.value.isascii():
+            return 127
+        # ... or at least Latin-1
+        try:
```

Comment:
I'm not convinced that trying to encode is simpler than just going from `max(self.value)`

Reply:
Hmm. True. That was older code that I copied together and simplified. I found the `max(self.value)` bit but didn't think of just applying it everywhere. I'll cut it down, thanks.